### PR TITLE
Allow only positive non-decimal values in duration schema

### PR DESCRIPTION
**Pull request by HLeithner**
Pull Request for Issue #41399 .

### Summary of Changes
Minimum set to 0 and only numbers are allowed

### Testing Instructions
Add a recipe and try a negative and or decimal duration

### Actual result BEFORE applying this Pull Request
nothing

### Expected result AFTER applying this Pull Request
marked invalid on blur

### CODE REVIEW

**Comment by brianteeman** [comment]
No change

image

**Comment by brianteeman** [comment]
pattern does not work on number fields

**Comment by brianteeman** [comment]
also you have told me many times in the past that clientside only validation is not good enough

**Comment by HLeithner** [comment]
> pattern does not work on number fields

Interesting, j read it doesn't work in chrome, I tested it in Vivaldi and thought it's fixed for all chromium based browsers.

Also interesting that joomla doesn't validate it patterns, have to check this, don't we have a regex validation?

**Comment by brianteeman** [comment]
i've lost track now of the number of times over the years I have raised that forms are not being validated when we think they are and vice versa.

**Comment by brianteeman** [comment]
> > pattern does not work on number fields
> 
> Interesting, j read it doesn't work in chrome, I tested it in Vivaldi and thought it's fixed for all chromium based browsers.
> 

Nothing to do with chrome. its part of the specification

> the pattern attribute is an attribute of the [text](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/text), [tel](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/tel), [email](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email), [url](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/url), [password](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/password), and [search](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search) input types.
https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/pattern

**Comment by dgrammatiko** [comment]
> Also interesting that joomla doesn't validate it patterns

Validation per field is OPT-IN, not default...

File: plugins/schemaorg/recipe/forms/duration.xml [hunk 1]
```diff
@@ -4,12 +4,16 @@
         name="hour"
         type="number"
         label="PLG_SCHEMAORG_RECIPE_FIELD_HOUR_LABEL"
+		min="0"
+		pattern="\d+"
```
Comment (richard67) [inline plugins/schemaorg/recipe/forms/duration.xml:8]:
```suggestion
        min="0"
        pattern="\d+"
```
Use same indentation as for previous line.

File: plugins/schemaorg/recipe/forms/duration.xml [hunk 2]
```diff
@@ -4,12 +4,16 @@
...
     />
 
     <field
         name="min"
         type="number"
         label="PLG_SCHEMAORG_RECIPE_FIELD_MINUTE_LABEL"
-        max="59"
+        min="0"
+		max="59"
+		pattern="\d+"
```
Comment (richard67) [inline plugins/schemaorg/recipe/forms/duration.xml:17]:
```suggestion
        min="0"
        max="59"
        pattern="\d+"
```
Use same indentation as for previous line.

Comment (brianteeman) [reply plugins/schemaorg/recipe/forms/duration.xml:17]: the problem is that the file is space indented (it should be tab indented) but these changes are tab indented

Comment (richard67) [reply plugins/schemaorg/recipe/forms/duration.xml:17]: Yes, I saw. I just think it should be fixed for the complete file separately. But anyway the PR seems to have other, more important problems.

Comment (brianteeman) [reply plugins/schemaorg/recipe/forms/duration.xml:17]: not just this pr but the whole schemaorg but no one seems to care

**Comment by HLeithner** [comment]
you are right that pr doesn't solve the issue, mainly because we don't have a correct field for this. In theory we have a time field with time rule which would fit, with the exception that the maximum could only be 24 hours and can't be easily extended to more then 99 ours...

Also we are missing a simple regex rule... 

I'm closing this pr. sorry for wasting your time.

**Pull request closed without merging**